src/kvdb/badger_database.go

```diff
@@ -62,19 +62,33 @@ func (w *BadgerDatabase) Get(key []byte) (res []byte, err error) {
 		return rerr
 	})
 
+	if err == badger.ErrKeyNotFound {
+		err = nil
+	}
 	return
 }
 
 // Delete removes key-value pair by key.
 func (w *BadgerDatabase) Delete(key []byte) error {
 	tx := w.db.NewTransaction(true)
-	defer tx.Discard()
```

Comment:
Keep the defer, wrap your new logic in a lambda on it.

Reply:
Agreed, I don't know where that came from.